django/db/models/expressions.py

```diff
@@ -1303,14 +1303,19 @@ def as_sql(self, compiler, connection, template=None, **extra_context):
         # WHERE clause, but in a CASE WHEN expression they must use a predicate
         # that's always True.
         if condition_sql == "":
-            condition_sql, condition_params = compiler.compile(Value(True))
+            if connection.features.supports_boolean_expr_in_select_clause:
+                condition_sql, condition_params = compiler.compile(Value(True))
+            else:
+                condition_sql, condition_params = "1=1", ()
```

Comment:
😮 

I see that we have a similar condition here `django.db.models.expressions.Exists.as_sql`:
```
                features = compiler.connection.features
                if not features.supports_boolean_expr_in_select_clause:
                    return "1=1", ()
                return compiler.compile(Value(True))
```

Would it make sense to somehow move them to the same piece of logic?
We could either have `compile(Value(True))` determine if it should generate a `1=1`, but it have other side-effects I think 😕

Reply:
It could be difficult because this caveat is quite complicated, e.g. you cannot use `boolean` in the `SELECT` clause, but you can use in the `WHERE` clause, as for `CASE WHEN` you cannot use `boolean` in both places :smile: everything depends ... :disappointed:

Reply:
That's what I thought.
Thanks for the clarification 👍 The patch is fine to me 🚀